src/pybind/mgr/volumes/fs/subvolume.py

```diff
@@ -111,11 +111,21 @@ def create_subvolume(self, spec, size=None,  pool=None, namespace_isolated=True,
                 uid = int(groupstat.st_uid)
             elif uid < 0:
                 raise VolumeException(-errno.EINVAL, "Provide a valid uid")
+            else:
+                try:
+                    uid = int(uid)
+                except ValueError:
+                    raise VolumeException(-errno.EINVAL, "Invalid uid")
```

Comment:
This is missing the `uid < 0` sanity check above. Suggest you reorder the logic.

Reply:
I have tried that, but `isinstance(uid, basestring)` and `isinstance(uid, str)` doesn't work as expected in teuthology runs. I will check if there is an alternative.